src/dmrf/webpackModules/dmrf.ts

```diff
@@ -2,51 +2,90 @@ import spacepack from "@moonlight-mod/wp/spacepack_spacepack";
 import Dispatcher from "@moonlight-mod/wp/discord/Dispatcher";
 import { DMRFNatives } from "../types";
 
-const COOL = "Queueing message to be sent";
-const module = spacepack.findByCode(COOL)[0].exports;
+const MessageActionCreators = spacepack.require(
+  "discord/actions/MessageActionCreators"
+).default;
 
 const natives: DMRFNatives = moonlight.getNatives("dmrf");
 const logger = moonlight.getLogger("dmrf");
 
 natives.init();
 
-const originalSend = module.Z.sendMessage;
-module.Z.sendMessage = async (...args: any[]) => {
-  logger.trace("got sendMessage");
-  const message = args[1];
-  logger.trace("handling sendMessage", message);
+const originalSend = MessageActionCreators.sendMessage;
+const originalEdit = MessageActionCreators.editMessage;
+
+async function hook(message: any, name: string): Promise<boolean | any> {
+  logger.debug(`Hooking ${name}`);
   const result = await natives.sendHook(message);
-  logger.trace("handled", result);
+  logger.debug(`${name} - Result:`, result);
   if (result == null) {
-    logger.error("dropping on sendHook not supported yet");
-    return;
+    // NOTE(cyn): this error seems redundant now?
+    logger.error(`${name} - Dropping on sendHook not supported yet`);
+    return false;
   }
+
+  return result;
+}
+
+MessageActionCreators.sendMessage = async function (
+  ...args: any[]
+): Promise<any> {
+  const result = await hook(args[1], "sendMessage");
+  if (result === false) return;
   args[1] = result;
-  return originalSend.call(module.Z, ...args);
+
+  return originalSend.call(MessageActionCreators, ...args);
 };
 
+MessageActionCreators.editMessage = async function (
```

Comment:
as talked about in private, could this behavior become a setting (turned on by default)?